src/main/java/com/epam/aidial/core/security/ExtractedClaims.java

```diff
@@ -1,6 +1,7 @@
 package com.epam.aidial.core.security;
 
 import java.util.List;
+import java.util.Map;
 
-public record ExtractedClaims(String sub, List<String> userRoles, String userHash) {
+public record ExtractedClaims(String sub, List<String> userRoles, String userHash, Map<String, List<String>> userClaims) {
```

Comment:
Perhaps we can put user claims  reference and later a caller may extract some claims if needed.
What do you think?

Reply:
Moreover I would move the logic of some claims extraction to the class ExtractedClaims.
and leave just userHash since it's computed.

Reply:
as agreed. let's postpone the suggested implementation